### PR TITLE
[UI][Admin] Form menu builders for Product and Product Variant edits

### LINKED ISSUE #7509: Admin tabs error messages
Hey,

It may seems useless, but I think it's an important enhancement to show which tab have an error to the user.
The problem is that sylius didn't add a "tab manager" to the forms, the tabs are only twig templates.

Maybe we could  add a sonata like tab form management ?
Or a dirty working fix, could be a js solution (check if tab has error messages, if yes show some icon) ?

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | yes |
| Related tickets | fixed #7509 (by accident) |
| License         | MIT |

Moved `product` and `product variant`'s menus generation to KNP's `MenuBuilder`.

And while at it, I've gone waltzing matilda for a while and added error indicators to tab/accordion-based forms.

Tabs:
tabs

Accordion:
accordion

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Menu/ProductFormMenuBuilder.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        $this->factory = $factory;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * @return ItemInterface
+     */
+    public function createMenu()
+    {
+        $menu = $this->factory->createItem('root');
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AdminBundle/Menu/ProductFormMenuBuilder.php]: Would be cool to have access to the actual product here, to decide what kind of sections to display. WDYT? Have a look at @GSadee's PR.

File: src/Sylius/Bundle/AdminBundle/Menu/ProductFormMenuBuilder.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        $menu
+            ->addChild('details')
+            ->setAttribute('data-tab', 'details')
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AdminBundle/Menu/ProductFormMenuBuilder.php]: This should be derived from the menu item name and used in template to render properly. Does not make sense to redefine it here.

File: src/Sylius/Bundle/UiBundle/Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -855,6 +856,7 @@ sylius:
         values: 'Values'
         variant: 'Variant'
         variant_selection_method: 'Variant selection method'
+        variant_choice: 'Variant choice'
```
Comment (GSadee) [inline src/Sylius/Bundle/UiBundle/Resources/translations/messages.en.yml]: This line should be above `variant_selection_method `

**Comment by pjedrzejewski** [comment]
Thanks Jasiek, could you also update the documentation, please?

**Comment by NeverResponse** [comment]
On it.

**Pull request merged**